src/js/db/DBD.js

```diff
@@ -0,0 +1,179 @@
+
+				// Check if type is valid, otherwise error out
+				if(!validTypes.includes(type))
+					throw new Error('Unsupported DBD type: ' + type);
+
+				// SKIPPED: Foreign key reading
+
+				// NAME READING
+				const firstSpace = line.indexOf(' ');
+				const lastSpace = line.lastIndexOf(' ');
+				const name = line.substring(firstSpace + 1, (firstSpace === lastSpace) ? line.length : line.indexOf(' ', firstSpace + 1)).replace('?', '');
+
+				// SKIPPED: Verified (?) and comment reading
+
+				columnDefinitions[name] = type;
+			}
+		}
+
+		let layouthashes = [];
+		let builds = [];
+		let fields = [];
+
+		for(let i = lineNumber; i < lines.length; i++){
+			// Line is manipulated later on, not a const
+			let line = lines[i];
+
+			if(line.length === 0){
+				versionDefinitions.push({fields, builds, layouthashes});
+
+				layouthashes = [];
+				builds = [];
+				fields = [];
+			}
+
+			// Parse layouthashes
+			if(line.substring(0, 6) === "LAYOUT"){
+				layouthashes = line.replace("LAYOUT ", "").split(", ");
+			}
+
+			// Parse builds, skip BUILD lines with ranges
+			if(line.substring(0, 5) === "BUILD" && !line.includes('-')){
+				builds = builds.concat(builds, line.replace("BUILD ", "").split(", "));
+			}
+
+			if (line.substring(0, 6) !== "LAYOUT" && line.substring(0, 5) !== "BUILD" && line.substring(0, 7) !== "COMMENT" && line.length !== 0)
+			{
+				let field = [];
+				field.isID = false;
+				field.isNonInline = false;
+				field.isRelation = false;
+				field.arrLength = 0;
+
+				if(line.indexOf('$') !== -1){
+					// Annotation
+					const start = line.indexOf('$');
+					const end = line.indexOf('$', 1);
+
+					const annotations = line.substring(start + 1, end - 1).split(",");
+
+					if(annotations.includes("id"))
+						field.isID = true;
+
+					if(annotations.includes("noninline"))
+						field.isNonInline = true;
+
+					if(annotations.includes("relation"))
+						field.isRelation = true;
+
+					line = line.substring(end + 1);
+				}
+
+				if(line.indexOf('<') !== -1){
+					// Signedness, cardinality
+					const start = line.indexOf('<');
+					const end = line.indexOf('>');
+					let size = line.substring(start + 1, end);
+					if(size[0] === 'u'){
+						field.isSigned = false;
+						field.size = size.replace('u', '');
+						line = line.replace("<u" + field.size + ">", "");
+					}else{
+						field.isSigned = true;
+						field.size = size;
+						line = line.replace("<" + field.size + ">", "");
+					}
+				}
+
+				if(line.indexOf('[') !== -1){
+					// Array length
+					const start = line.indexOf('[');
+					const end = line.indexOf(']');
+					field.arrLength = line.substring(start + 1, end);
+					line = line.replace("[" + field.arrLength + "]", "");
+				}
+
+				if(line.indexOf("//") !== -1){
+					// Comment, skip
+					line = line.substring(0, line.indexOf("//"));
+				}
+
+				field.name = line;
+
+				if(!(field.name in columnDefinitions))
```

Comment:
Replace with `if (columnDefinitions.hasOwnProperty(field.name))` if `columnDefinitions` becomes an object, or better `if (columnDefinitions.has(field.name))` if it becomes a `Map()`.